### PR TITLE
fix: resolve iOS touch blur issue

### PR DESCRIPTION
**Pull request by aMollusk**
**What**: #426 

**Why**: To ensure consistent behaviour across devices.

**How**: Adding a `touchstart` event listener to check if the target is anything but downshift.

**Checklist**:

* [ ] Documentation N/A
* [x] Tests
* [ ] Ready to be merged
      
* [x] Added myself to contributors table

### CODE REVIEW

**Comment by aMollusk** [comment]
WIP - This hasn't been fully tested.

File: src/downshift.js [hunk 1]
```diff
@@ -881,13 +881,28 @@ class Downshift extends Component {
...
         }
       }
+      // Touching an element in iOS gives focus and hover states, but touching out of
+      // the element will remove hover, and persist the focus state, resulting in the
+      // blur event not being triggered.
+      const onTouchStart = event => {
+        const targetInDownshift =
+          this._rootNode && isOrContainsNode(this._rootNode, event.target)
+        if (!targetInDownshift && this.getState().isOpen) {
+          this.reset({type: Downshift.stateChangeTypes.mouseUp}, () =>
```
Comment (aMollusk) [inline src/downshift.js]: Not sure I fully grasp the concept of these `stateChangeTypes`. Should I be creating a `touchstart` `stateChangeType` property?

Comment (kentcdodds) [reply src/downshift.js]: Yes, please do. It's to allow folks to know when state changed and why. Used in stateReducer and onStateChange

Comment (aMollusk) [reply src/downshift.js]: Great. Thanks.

**Pull request merged**